### PR TITLE
Add new reference state data types to shared state

### PR DESCRIPTION
**Pull request by javierbrk**
Network reference state will be defined by the UI. The data types must be supported and sync using shared state. This package provides two new data types one for WiFi links information and the other for nodes available in the network

### CODE REVIEW

File: packages/shared-state-reference-state/Makefile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+PKG_NAME:=shared-state-wifi_links_info
+PKG_VERSION=$(GIT_COMMIT_DATE)-$(GIT_COMMIT_TSTAMP)
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/$(PKG_NAME)
+	TITLE:=Reference state datatypes module for shared-state
+	CATEGORY:=LibreMesh
+	URL:=http://libremesh.org
+	DEPENDS:= shared-state
```
Comment (G10h4ck) [inline packages/shared-state-reference-state/Makefile:22]: are you sure it is ok with the space after `:=` anyway better get rid of it

File: packages/shared-state-reference-state/Makefile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#
+# This is free software, licensed under the GNU Affero General Public License v3.
+#
+
+include $(TOPDIR)/rules.mk
+
+GIT_COMMIT_DATE:=$(shell git log -n 1 --pretty=%ad --date=short . )
+GIT_COMMIT_TSTAMP:=$(shell git log -n 1 --pretty=%at . )
+
+PKG_NAME:=shared-state-wifi_links_info
```
Comment (G10h4ck) [inline packages/shared-state-reference-state/Makefile:13]: Is the package name still appropriate?

**Review by G10h4ck — CHANGES_REQUESTED** [review]
Maybe this should be integrated directly into lime app package?

Plus see inline comments

**Comment by ilario** [comment]
Hi!
Two questions:
* would it be possible to use https://github.com/libremesh/lime-packages/blob/master/libremesh.mk for the Makefile? For example like here: https://github.com/libremesh/lime-packages/blob/e64db16c4e80f29b4687ad346c0e8d7833f7b62d/packages/lime-proto-bmx7/Makefile 
* this code is not specific for LibreMesh, so it should be on OpenWrt's repositories, not here. Are there plans for migrating this growing amount of shared-state code to OpenWrt?

**Comment by G10h4ck** [comment]
Yeah using [libremesh.mk](https://github.com/libremesh/lime-packages/blob/master/libremesh.mk) is in usually a good idea, anyway I have talked with @javierajorge and we concluded the two uci-defaults files that would be the only real addition from this package should instead be included in lime-app as the reference state stuff is all lime-app specific, so this PR will be closed once @javierbrk and @selankon include the contributed code in the proper package (lime-app os some of it's modules).

about upstreaming shared-state code to OpenWrt it is a good idea but not for now, ATM some parts of it like the neighbour candidate discovery does assumptions that are quite tied to libremesh network model even if it doesn't depends on it directly, so we would risk to upstream something that works badly on a different setup, I am working now on a new version of shared-stare that will be a more reasonable candidate for upstreaming to OpenWrt.

Cheers!

File: packages/shared-state-reference-state/Makefile (hunk 2 above)
Comment (javierbrk) [reply packages/shared-state-reference-state/Makefile]: ups copy paste, this make file is no longer nedded

**Comment by javierbrk** [comment]
Here is an attempt to move the scripts into lime app package, what do you think ?

**Comment by G10h4ck** [comment]
> Here is an attempt to move the scripts into lime app package, what do you think ?

Seems fine to me, let's wait to hear @selankon if he thinks this is the proper place where to ship those uci-defaults

**Review by G10h4ck — APPROVED** [review]

**Pull request merged**